crates/sui-graphql-rpc/src/types/transaction_block/tx_lookups.rs

```diff
@@ -0,0 +1,468 @@
+        format!(
+            "tx_sequence_number >= {} AND tx_sequence_number <= {}",
+            bound.lo, bound.hi
+        )
+    )
+}
+
+pub(crate) fn select_pkg(
+    pkg: &SuiAddress,
+    sender: Option<SuiAddress>,
+    bound: TxBounds,
+) -> RawQuery {
+    filter!(
+        select_tx(sender, bound, "tx_calls_pkg"),
+        format!("package = {}", bytea_literal(pkg))
+    )
+}
+
+pub(crate) fn select_mod(
+    pkg: &SuiAddress,
+    mod_: String,
+    sender: Option<SuiAddress>,
+    bound: TxBounds,
+) -> RawQuery {
+    filter!(
+        select_tx(sender, bound, "tx_calls_mod"),
+        format!("package = {} and module = {{}}", bytea_literal(pkg)),
+        mod_
+    )
+}
+
+pub(crate) fn select_fun(
+    pkg: &SuiAddress,
+    mod_: String,
+    fun: String,
+    sender: Option<SuiAddress>,
+    bound: TxBounds,
+) -> RawQuery {
+    filter!(
+        select_tx(sender, bound, "tx_calls_fun"),
+        format!(
+            "package = {} AND module = {{}} AND func = {{}}",
+            bytea_literal(pkg),
+        ),
+        mod_,
+        fun
+    )
+}
+
+pub(crate) fn select_kind(kind: TransactionBlockKindInput, bound: TxBounds) -> RawQuery {
+    filter!(
+        select_tx(None, bound, "tx_kinds"),
+        format!("tx_kind = {}", kind as i16)
+    )
+}
+
+pub(crate) fn select_sender(sender: &SuiAddress, bound: TxBounds) -> RawQuery {
+    select_tx(Some(*sender), bound, "tx_senders")
+}
+
+pub(crate) fn select_recipient(
+    recv: &SuiAddress,
+    sender: Option<SuiAddress>,
+    bound: TxBounds,
+) -> RawQuery {
+    filter!(
+        select_tx(sender, bound, "tx_recipients"),
+        format!("recipient = '\\x{}'::bytea", hex::encode(recv.into_vec()))
+    )
+}
+
+pub(crate) fn select_input(
+    input: &SuiAddress,
+    sender: Option<SuiAddress>,
+    bound: TxBounds,
+) -> RawQuery {
+    filter!(
+        select_tx(sender, bound, "tx_input_objects"),
+        format!("object_id = '\\x{}'::bytea", hex::encode(input.into_vec()))
+    )
+}
+
+pub(crate) fn select_changed(
+    changed: &SuiAddress,
+    sender: Option<SuiAddress>,
+    bound: TxBounds,
+) -> RawQuery {
+    filter!(
+        select_tx(sender, bound, "tx_changed_objects"),
+        format!(
+            "object_id = '\\x{}'::bytea",
+            hex::encode(changed.into_vec())
+        )
+    )
+}
+
+pub(crate) fn select_ids(ids: &Vec<Digest>, bound: TxBounds) -> RawQuery {
+    let query = select_tx(None, bound, "tx_digests");
+    if ids.is_empty() {
+        filter!(query, "1=0")
+    } else {
+        let mut inner = String::new();
+        let mut prefix = "tx_digest IN (";
+        for id in ids {
+            write!(
+                &mut inner,
+                "{prefix}'\\x{}'::bytea",
+                hex::encode(id.to_vec())
+            )
+            .unwrap();
+            prefix = ", ";
+        }
+        inner.push(')');
+        filter!(query, inner)
+    }
+}
+
+pub(crate) fn bytea_literal(addr: &SuiAddress) -> impl fmt::Display + '_ {
```

Comment:
There are a couple of other places where you are creating bytea literals from things that are not `SuiAddress`es -- maybe you should make this more general, i.e. accept a `&[u8]`? You should be able to get such a slice out of any of the things that you want to represent this way.

Reply:
Ah yes... just ran into this on events and realized the "mistake" I had made. I'll do this